src/services/assetService.ts

```diff
@@ -151,11 +151,14 @@ export class AssetService {
     public async save(metadata: IAssetMetadata): Promise<IAssetMetadata> {
         Guard.null(metadata);
 
+        const fileName = `${metadata.asset.id}${constants.assetMetadataFileExtension}`;
+
         // Only save asset metadata if asset is in a tagged state
         // Otherwise primary asset information is already persisted in the project file.
         if (metadata.asset.state === AssetState.Tagged) {
-            const fileName = `${metadata.asset.id}${constants.assetMetadataFileExtension}`;
             await this.storageProvider.writeText(fileName, JSON.stringify(metadata, null, 4));
+        } else {
+            await this.storageProvider.deleteFile(fileName);
```

Comment:
why are we deleting file here?

Reply:
If the asset is not in a tagged state then there is no reason to keep it around since it doesn't contain and region data.